packages/async-loader/async.js

```diff
@@ -1,36 +1,80 @@
-import { h, Component } from 'preact';
+// import { options, h, Component } from 'preact';
+import { h, Component, options, render } from 'preact';
 
-export default function(req) {
-	function Async() {
-		Component.call(this);
+const f = 'unmount';
+const oldUnmountOpts = options[f];
+const oldDiffed = options.diffed;
 
-		let b, old;
-		this.componentWillMount = () => {
-			b = this.base = this.nextBase || this.__b; // short circuits 1st render
-			req(m => {
-				this.setState({ child: m.default || m });
-			});
-		};
+options[f] = function(vnode) {
+	/**
+	 *  3. now Pending is being unmount and its dom is being removed
+	 *  so we detach the dom from the vnode and empty its children so that Pending is unmount
+	 *  but the DOM is never actually removed.
+	 */
+	if (vnode.type === Pending) {
+		vnode.__e = null;
```

Comment:
maybe we should have these as a map or just mapping vars

```js
const keys = {
	dom: '__e',
	children: '__k',
	...
};
```

```js
const dom = '__e';
const children = '__k';
...
```

Reply:
The constants would be nice since they'd get inlined.

Reply:
done